packages/web-extension/test/remoteApi/remoteApi.integration.test.ts

```diff
@@ -438,6 +451,22 @@ describe('remoteApi integration', () => {
           }, 1);
         });
 
+        it('does not emit if source was disabled', (done) => {
+          someNumbersSource$.next(1n);
+          const emittedBeforeDisable = firstValueFrom(sut.consumer.someNumbers$);
+          setTimeout(async () => {
+            expect(await emittedBeforeDisable).toEqual(1n);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            sut.api$.next(null as any);
+            setTimeout(async () => {
+              const emittedAfterDisable = firstValueFrom(race(sut.consumer.someNumbers$, of('nothing').pipe(delay(1))));
+              setTimeout(() => sut.hostMessenger.shutdown());
```

Comment:
Is `setTimeout` really needed here?

Reply:
No it isn't. Removed it: [307145f](https://github.com/input-output-hk/cardano-js-sdk/pull/556/commits/307145fec385223f7d8ca9fc3d46c89ba921aa4c)